### PR TITLE
Fix docker dependencies

### LINKED ISSUE #33657: Use conditional PR tag for docker images
[docker_image_check](https://github.com/ClickHouse/ClickHouse/blob/92099302e79c18a9993e0ad22aee408f9c92b406/tests/ci/docker_images_check.py#L132) currently always use a custom tag for our child images. It causes failed dependencies from time to time, for example in #24258 

https://s3.amazonaws.com/clickhouse-test-reports/24258/c1c3f25c10ceb467fe4906c5772d1e94344aaaaf/push_to_dockerhub__actions__aarch64.html

We know if the parent image is built or not, so it's easy to distinguish, if custom tag should be used

ccL @tavplubix

### PR DESCRIPTION
**Pull request by Felixoid**
Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Fix unconditional `--build-arg FROM_TAG=` docker argument
- Add some tests for docker tests/ci

### CODE REVIEW

**Comment by Felixoid** [comment]
Docker builds work good now https://github.com/ClickHouse/ClickHouse/runs/4859937340?check_suite_focus=true

File: tests/ci/docker_images_check.py [hunk 1]
```diff
@@ -24,21 +24,49 @@
...
+        self,
+        path: str,
+        repo: str,
+        parrent: Optional["DockerImage"] = None,
+        gh_repo_path: str = GITHUB_WORKSPACE,
+    ):
+        self.path = path
+        self.full_path = os.path.join(gh_repo_path, path)
+        self.repo = repo
+        self.parrent = parrent
```
Comment (alesapin) [inline tests/ci/docker_images_check.py:38]:
```suggestion
        self.parent = parent
```

Comment (alesapin) [reply tests/ci/docker_images_check.py:38]: And in all other places :)

**Review by alesapin — APPROVED** [review]
Much better, but need to replace `parrent` with `parent` :)

**Comment by alexey-milovidov** [comment]
@alesapin, I don't get it.

- What is the motivation for this change?
- What is the difference between images.json and docker_images.json?
- What the test is doing?

There is no explanation anywhere, even in the description of this PR.

**Comment by Felixoid** [comment]
Motivation for fixing or for tests?

The first one fixes the proper usage of the source image and builds it before the dependent image.

The `tests/ci/tests/docker_image.json` is there to have a static test set of images. Otherwise, tests would fail after each change.

The test itself checks different places related to docker building/merging places. Like composing the build tree based on changed files, and producing a correct `docker build` commands

**Comment by qoega** [comment]
>tests/ci/tests/docker_image.json

You can rename it to make it clear. Like `for_ut_docker_image_snapshot.json`

**Comment by Felixoid** [comment]
Good idea, done in #52869

**Pull request merged**